src/Shared/Model/Metadata/NuGetPackageVersionMetadata.cs

```diff
@@ -0,0 +1,145 @@
+    public long? DownloadCount { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.IconUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri IconUrl { get; init; }
+
+    [JsonIgnore]
+    public PackageIdentity Identity => new(Name, new NuGetVersion(Version));
+
+    [JsonProperty(PropertyName = JsonProperties.Version)]
+    public string Version { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.LicenseUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri LicenseUrl { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.ProjectUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri ProjectUrl { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.ReadmeUrl)]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri ReadmeUrl { get; init; }
+
+    [JsonIgnore]
+    public Uri ReportAbuseUrl { get; }
+
+    [JsonProperty(PropertyName = "packageDetailsUrl")]
+    [JsonConverter(typeof(SafeUriConverter))]
+    public Uri PackageDetailsUrl { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.Published)]
+    public DateTimeOffset? Published { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.Owners)]
+    [JsonConverter(typeof(MetadataFieldConverter))]
+    public string Owners { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.PackageId)]
+    public string Name { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.RequireLicenseAcceptance, DefaultValueHandling = DefaultValueHandling.Populate)]
+    [DefaultValue(false)]
+    [JsonConverter(typeof(SafeBoolConverter))]
+    public bool RequireLicenseAcceptance { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.Summary)]
+    public string Summary { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.Tags)]
+    [JsonConverter(typeof(MetadataFieldConverter))]
+    public string Tags { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.Title)]
+    public string Title { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.Listed)]
+    public bool IsListed { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.PrefixReserved)]
+    public bool PrefixReserved { get; init; }
+
+    [JsonIgnore]
+    public LicenseMetadata LicenseMetadata { get; }
+
+    [JsonProperty(PropertyName = JsonProperties.Vulnerabilities)]
+    public IEnumerable<PackageVulnerabilityMetadata> Vulnerabilities { get; init; }
+
+    [JsonProperty(PropertyName = JsonProperties.SubjectId)]
+    public Uri CatalogUri { get; init; }
+
+    /// <summary>
+    /// Initialize an instance of <see cref="NuGetPackageVersionMetadata"/> using the <see cref="JsonConstructorAttribute"/>.
+    /// </summary>
+    /// <remarks>Necessary for unit test implementation of json serialization and deserialization.</remarks>
+    [JsonConstructor]
+#pragma warning disable CS8618
+    public NuGetPackageVersionMetadata()
```

Comment:
So based on experimenting, and there could be another way of doing this, I believe I have to have this empty constructor in order to deserialize json. I can make it private and initialize all values as `null!` or set them to all be nullable to avoid the warning. But without this constructor, the unit tests at the json deserialization step fail.